pages/features/cache.mdx

```diff
@@ -0,0 +1,110 @@
+
+FlakeHub Cache is currently in **private beta** but you can sign up for the beta here and we'll add you as soon as we're ready:
+
+<SignupForm />
+
+## Access control
+
+Nix caching solutions based on [nix-serve] serve an entire [Nix store][store] as a single monolithic cache and make all store paths in that cache available to anyone with a public key.
+With FlakeHub Cache, on the contrary, you can grant or deny read access to specific users **at the [flake][flakes] level**, which affords substantially more granular access control.
+
+Let's say that you're an admin for the org `omnicorp`.
+You could manage access to your org's flakes in such a way that, for example:
+
+* Only some trusted users would be allowed to pull store paths associated with the flake `super-duper-secure`.
+* Anyone in your org can pull store paths associated with the `dev-environments` flake.
+
+## Authentication model
+
+FlakeHub Cache authentication is based on [JSON Web Tokens][jwt] (JWTs or "jots").
+We currently use [GitHub] as our JWT authentication provider but will be adding support for [GitLab] in the near future.
+Down the road, the flexibility of JWTs will enable us to support a wide range of additional authentication solutions, including [SAML] and [Single sign-on][sso] (SSO) providers.
+
+In contrast, existing Nix caching services use static tokens for authentication.
+This model places all of the operational burden of authentication on users; with FlakeHub Cache, however, we've opted to shift that burden to trusted authentication providers.
+
+## Pushing to the cache
+
+FlakeHub Cache differs from existing Nix caching solutions because it forbids pushing to the cache in an ad-hoc way.
+Thus, you can't push to the cache from your laptop or from one of your servers.
+Instead, only *trusted builders* are allowed to push to the cache.
+At the moment, this includes only [GitHub Actions runners][actions] but other options will be available in the future.
+
+In order to push to FlakeHub Cache from GitHub Actions, your GitHub organization needs to be [registered for the beta](#signup).
+Once you've been added&mdash;you'll be notified via email&mdash;you can configure your org's GitHub Actions workflows to use the cache.
+Here's an example:
+
+```yaml {4-6,9-10} showLineNumbers filename=".github/workflows/nix.yml"
+jobs:
+  build-nix-package:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+      # Build default package
+      - run: nix build
+```
+
+In this configuration:
+
+* The `permissions` section enables you to authenticate to FlakeHub Cache.
+  No other authentication method, like public keys, is necessary.
+* The [Magic Nix Cache Action][mnca]
+
+<Callout type="info">
```

Comment:
Not clear how this callout relates to the stuff above it. Probably we want to say something like "You can push your stuff to the cache using one of the following actions: `magic-nix-cache` and Determinate CI."